pkg/ddl/index.go

```diff
@@ -979,31 +977,25 @@ func runIngestReorgJob(w *worker, d *ddlCtx, t *meta.Meta, job *model.Job,
 		return false, ver, errors.Trace(err)
 	}
 	done, ver, err = runReorgJobAndHandleErr(w, d, t, job, tbl, allIndexInfos, false)
-	if err != nil {
-		if !errorIsRetryable(err, job) {
-			logutil.DDLLogger().Warn("run reorg job failed, convert job to rollback",
-				zap.String("job", job.String()), zap.Error(err))
-			ver, err = convertAddIdxJob2RollbackJob(d, t, job, tbl.Meta(), allIndexInfos, err)
-		}
-		return false, ver, errors.Trace(err)
-	}
-	if !done {
-		return false, ver, nil
-	}
-
-	err = bc.FinishImport(tbl)
 	if err != nil {
 		if kv.ErrKeyExists.Equal(err) {
 			logutil.DDLLogger().Warn("import index duplicate key, convert job to rollback", zap.Stringer("job", job), zap.Error(err))
 			ver, err = convertAddIdxJob2RollbackJob(d, t, job, tbl.Meta(), allIndexInfos, err)
+		} else if !errorIsRetryable(err, job) {
+			logutil.DDLLogger().Warn("run reorg job failed, convert job to rollback",
+				zap.String("job", job.String()), zap.Error(err))
+			ver, err = convertAddIdxJob2RollbackJob(d, t, job, tbl.Meta(), allIndexInfos, err)
 		} else {
-			logutil.DDLLogger().Warn("lightning import error", zap.Error(err))
+			logutil.DDLLogger().Warn("run add index ingest job error", zap.Error(err))
 			if !errorIsRetryable(err, job) {
```

Comment:
why here is another `errorIsRetryable`? We already have line 984

Reply:
I forgot to remove one of `errorIsRetryable` when I merged the error handling logic of `FinishImport()` and `runReorgJobAndHandleErr()`..

Reply:
https://github.com/pingcap/tidb/pull/54149/commits/559aab2774b4a9a2de6be81d160c4a0159650ec6